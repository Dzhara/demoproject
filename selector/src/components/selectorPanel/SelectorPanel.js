import React from "react";
import PropTypes from "prop-types";
import CheckboxGroup from "../checkboxGroup/CheckboxGroup";
import CloseIcon from "../closeIcon/CloseIcon";
import style from "./SelectorPanel.module.scss";

export default class SelectorPanel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedItems: []
    };
  }

  onDeselected = value => {
    if (value === undefined) return;

    this.setState(prevState => {
      const newItems = prevState.selectedItems.slice();
      const index = prevState.selectedItems.indexOf(value);
      newItems.splice(index, 1);
      return Object.assign({}, prevState, { selectedItems: newItems });
    });

  };

  onSelected = (value) => {
    if (value === undefined) return;

    this.setState(prevState => {
      const newItems = prevState.selectedItems.slice();
      const index = prevState.selectedItems.indexOf(value);
      if (index === -1) {
        newItems.push(value);
      } else newItems.splice(index, 1);
      return Object.assign({}, prevState, { selectedItems: newItems });
    });

  };

  selectedItemTemplate = el => {
    return (
      <div
        className={style.selectedItem}
        key={`selected-item-${el}`}
      >
        {el}
        <CloseIcon onClick={() => {
          this.onDeselected(el);
        }} />
      </div>
    );
  };

  selectedItemsTemplate = () => {
    return (
      <div className={style.selectedItemsWrapper}>
        {
          this.state.selectedItems.map((el) => {
            return this.selectedItemTemplate(el);
          })
        }
      </div>
    );
  };

  availableItemsTemplate = () => {
    return (
      <div className={style.availableItemsWrapper}>
        <CheckboxGroup
          onSelected={this.onSelected}
          selectedItems={this.state.selectedItems}
          source={this.props.source}
        />
      </div>
    );
  };

  render() {
    return (
      <div className={style.panelContainer}>
        {this.selectedItemsTemplate()}
        {this.availableItemsTemplate()}
      </div>
    );
  }
}

SelectorPanel.propTypes = {
  source: PropTypes.arrayOf(Object)
};
