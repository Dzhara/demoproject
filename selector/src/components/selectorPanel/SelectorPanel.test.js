import React from "react";
import renderer from "react-test-renderer";
import SelectorPanel from "./SelectorPanel";

describe("SelectorPanel", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <SelectorPanel source={[{ label: "test", value: "test" }]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
