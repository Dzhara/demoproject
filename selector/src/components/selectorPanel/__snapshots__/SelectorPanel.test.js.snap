// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorPanel snapshot renders 1`] = `
<div
  className="panelContainer"
>
  <div
    className="selectedItemsWrapper"
  />
  <div
    className="itemsWrapper"
  >
    <div
      className="checkboxGroupWrapper"
    >
      <div>
        <div
          className="ui checkbox badge "
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            name={null}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label>
            test
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
